new-log-viewer/src/components/StatusBar/LogLevelSelect/index.tsx

```diff
@@ -0,0 +1,235 @@
+                        value={logLevelValue}
+                        onClick={onCheckboxClick}/>
+                </Tooltip>
+            </ListItemDecorator>
+            <Tooltip
+                placement={"left"}
+                title={
+                    <Stack
+                        alignItems={"center"}
+                        direction={"row"}
+                    >
+                        {logLevelName}
+                        {" and below"}
+                    </Stack>
+                }
+            >
+                <ListItemContent data-value={logLevelValue}>
+                    {logLevelName}
+                </ListItemContent>
+            </Tooltip>
+        </Option>
+    );
+};
+
+interface ClearFiltersOptionProps {
+    onClick: () => void
+}
+
+/**
+ * Renders an <Option/> to clear all filters in the <LogLevelSelect/>.
+ *
+ * @param props
+ * @param props.onClick
+ * @return
+ */
+const ClearFiltersOption = ({onClick}: ClearFiltersOptionProps) => (
+    <Option
+        value={null}
+        onClick={onClick}
+    >
+        <ListItemDecorator>
+            <CloseIcon/>
+        </ListItemDecorator>
+        Clear filters
+    </Option>
+);
+
+/**
+ * Renders a dropdown box for selecting log levels.
+ *
+ * @return
+ */
+const LogLevelSelect = () => {
+    const [selectedLogLevels, setSelectedLogLevels] = useState<LOG_LEVEL[]>([]);
+
+    const handleRenderValue = (selected: SelectValue<SelectOption<LOG_LEVEL>, true>) => (
+        <Box className={"log-level-select-render-value-box"}>
+            <Chip className={"log-level-select-render-value-box-label"}>
+                Log Level
+            </Chip>
+            {selected.map((selectedOption) => (
+                <LogLevelChip
+                    key={selectedOption.value}
+                    name={selectedOption.label as string}
+                    value={selectedOption.value}/>
+            ))}
+        </Box>
+    );
+
+    const handleCheckboxClick = useCallback((ev: React.MouseEvent<HTMLInputElement>) => {
+        ev.preventDefault();
+
+        const target = ev.target as HTMLInputElement;
+        const value = Number(target.value) as LOG_LEVEL;
+        let newSelectedLogLevels: LOG_LEVEL[];
+        if (selectedLogLevels.includes(value)) {
+            newSelectedLogLevels = selectedLogLevels.filter((logLevel) => logLevel !== value);
+        } else {
+            newSelectedLogLevels = [
+                ...selectedLogLevels,
+                value,
+            ];
+        }
+        setSelectedLogLevels(newSelectedLogLevels.sort((a, b) => a - b));
+    }, [selectedLogLevels]);
+
+    const handleSelectChange = useCallback((
+        ev: React.MouseEvent | React.KeyboardEvent | React.FocusEvent | null
+    ) => {
+        if (null === ev) {
+            setSelectedLogLevels([]);
+
+            return;
+        }
+
+        const target = ev.target as HTMLElement;
+        const selectedValue = Number(target.dataset.value);
```

Comment:
`selectedValue` will be `NaN` for the clear filters option path. Do you want to add the check to see if `target.dataset.value` exists here?